sqlalchemy_utils/functions/database.py

```diff
@@ -422,7 +422,13 @@ def is_auto_assigned_date_column(column):
 
 
 def _set_url_database(url: sa.engine.url.URL, database):
-    if hasattr(sa.engine, 'URL'):
+    """Set the database of an engine URL.
+
+    :param url: A SQLAlchemy engine URL.
+    :param database: New database to set.
+
+    """
+    if hasattr(url, 'set'):  # SQLAlchemy >1.4
```

Comment:
These still behave in the same way for me:

```
$ python
Python 3.8.5 (default, Jan 27 2021, 15:41:15) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sqlalchemy as sa
>>> sa.__version__
'1.3.23'
>>> hasattr(sa.engine, 'URL')
False
>>> from sqlalchemy.engine.url import make_url
>>> url = make_url('postgresql://myuser:***@localhost/mydb_test')
>>> hasattr(url, 'set')
False
>>> 
$ pip install 'sqlalchemy==1.4.3'
Collecting sqlalchemy==1.4.3
  Using cached SQLAlchemy-1.4.3-cp38-cp38-manylinux2014_x86_64.whl (1.5 MB)
Requirement already satisfied: greenlet!=0.4.17; python_version >= "3" in ./.venv/lib/python3.8/site-packages (from sqlalchemy==1.4.3) (1.0.0)
Installing collected packages: sqlalchemy
  Attempting uninstall: sqlalchemy
    Found existing installation: SQLAlchemy 1.3.23
    Uninstalling SQLAlchemy-1.3.23:
      Successfully uninstalled SQLAlchemy-1.3.23
Successfully installed sqlalchemy-1.4.3
$ python
Python 3.8.5 (default, Jan 27 2021, 15:41:15) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sqlalchemy as sa
>>> sa.__version__
'1.4.3'
>>> hasattr(sa.engine, 'URL')
True
>>> from sqlalchemy.engine.url import make_url
>>> url = make_url('postgresql://myuser:***@localhost/mydb_test')
>>> hasattr(url, 'set')
True
>>> 
```

Reply:
Ah wow. You are right. For some reason I was thinking >=1.4 should be returning False, but _that_ would be a bug haha.

So truly, only the dialect changes for postgresql are required. However, given that `hasattr('set')` is what the migration guides suggest, any reason you have a preference for the original URL check?

Reply:
But we cannot use `url.set()` (as explained earlier), what we actually use is `sa.engine.URL.create()`, hence the choice for `hasattr()`

Reply:
Yes, I did change it to use `URL.create` (https://github.com/nsoranzo/sqlalchemy-utils/pull/1/files#diff-afcd339dcee57062a3cffa0eb5a89652bb827e2eb36d7925396bf5699068dd62R432-R439)

But the check is still valid, no?

Reply:
I only make a case for it because it comes directly from SQLAlchemy's migration docs :smile: 
```python
def set_url_drivername(some_url, some_drivername):
    # check for 1.4
    if hasattr(some_url, "set"):
        return some_url.set(drivername=some_drivername)
    else:
        # SQLAlchemy 1.3 or earlier, mutate in place
        some_url.drivername = some_drivername
        return some_url
```

Reply:
But I also understand why you are saying it makes more since to check for `URL` (since that's what we are actually invoking),
so can change it back if you'd like.